components/brave_wallet_ui/common/hooks/use-multi-chain-buy-assets.ts

```diff
@@ -59,16 +61,16 @@ export const useMultiChainBuyAssets = () => {
     return selectedAsset ? getNetworkInfo(selectedAsset.chainId, selectedAsset.coin, buyAssetNetworks) : undefined
   }, [selectedAsset, buyAssetNetworks])
 
-  const selectedAssetBuyOptions = React.useMemo((): BuyOption[] => {
+  const selectedAssetBuyOptions: BuyOption[] = React.useMemo(() => {
+    const { wyreAssetOptions, rampAssetOptions, sardineAssetOptions } = options
+    const onRampAssetMap = {
+      [BraveWallet.OnRampProvider.kWyre]: wyreAssetOptions,
+      [BraveWallet.OnRampProvider.kRamp]: rampAssetOptions,
+      [BraveWallet.OnRampProvider.kSardine]: sardineAssetOptions
+    }
     return selectedAsset
-      ? BuyOptions.filter(buyOption => {
-          return isSelectedAssetInAssetOptions(
-            selectedAsset,
-            buyOption.id === BraveWallet.OnRampProvider.kWyre
-              ? options.wyreAssetOptions
-              : buyOption.id === BraveWallet.OnRampProvider.kRamp ? options.rampAssetOptions : []
-          )
-        })
+      ? BuyOptions.filter(buyOption => isSelectedAssetInAssetOptions(selectedAsset, onRampAssetMap[buyOption.id]))
+        .sort((optionA, optionB) => optionA.name.localeCompare(optionB.name))
```

Comment:
Is it possible to not do the sorting in-place, and not mutate the array on which `.sort()` is called?

Reply:
@onyb Created a new array before calling sort.